File: ts/9.palindrome-number.ts
/*
 * @lc app=leetcode id=9 lang=typescript
 *
 * [9] Palindrome Number
 */

// @lc code=start
function isPalindrome(x: number): boolean {
    if(x < 0){
        return false
    }
    let numStr: string = x.toString()
    if(numStr.length % 2 === 0){
       for(let i = 0; i < numStr.length/2; i++){
           if(numStr[i] != numStr[numStr.length - 1 - i]){
                return false
            }
       }
    }else{
        for(let i = 0; i < (numStr.length - 1)/2; i++){
            if(numStr[i] != numStr[numStr.length - 1 - i]){
                return false
            }
        }
    }
    return true
};
// @lc code=end

File: python/9.palindrome-number.py
class solution:
    def isPalindrome(x: int) -> bool:
        if x < 0:
            return False
        numStr = str(x)
        if len(numStr) % 2 == 0:
            for i in range(len(numStr)/2):
                if numStr[i] != numStr[len(numStr) - 1 - i]:
                    return False
        else:
            for i in range(int((len(numStr) - 1)/2)):
                if numStr[i] != numStr[len(numStr) - 1 - i]:
                    return False